### ISSUE TITLE
Justification Data Extraction

### ISSUE THREAD

**Issue by jmhernan** [skdreier/NIreland_NLP#2]
Data is in `.txt`, the output comes from nVivo. Each `.txt` the file contains individual justifications extracted from pdf scans and contains several tags including actual text.  These tags are:
`img_id`
`file_id`
`justification_text`

A sample entry looks like this:
```
Files\\\\DNV_round1\\\\DEFE 13 1358\\\\IMG_9711_DEFE_13_1358 - § 1 reference coded [ 4.20% 
Coverage]\nReference 1 - 4.20% Coverage\nInformation about the tribunal should emphasise the 
steps that will be \ntaken to separate the decision to detain a person from the executive.\n
```
## Approach
use `regex` to extract individual components into a dataframe.
1. Parse each File entry into rows.
`.*(?=Files)` Use "Files" as a seperator.
2. Extract each element row wise to address some inconsistencies that regex might not capture. Each element will be a new column. 
for image id use: `(IMG_\d{4})`
for doc id use: `(DEFE\w+|PREM|OTHERS...\w+)`
for justification: We can use something like "text between two anchors" or similar
3. clean up justification text.

**Comment by skdreier** [comment]
@jmhernan: Note for first step: Add a column indicating the name of the file each row/observation came from. This will signal what justification category each entry belongs to.

**Comment by skdreier** [comment]
We separated each entry (which represents a single document) into image, file, and text material and also retained the raw text for each document entry. We discovered that a given file could have several unique coding instances, each indicated with a reference number (e.g., "Reference 1"). 

Next steps include:
1. Adding a column indicating the justification category (corresponds with the .txt file name)
2. Parsing out each unique reference in a given page to the same code. (For example, a page which received 3 unique "Denial Justification" codes would be a single line with three text columns, each corresponding to a specific reference. (We also want to strip superfluous stuff (coverage percentages, etc.)  
3. Go back and retain the document texts without breaks per coded reference.
4. Loop through all the .txt justification files, parse each with this code, and append them to one another.
5. Fix those documents (PREM 15 1010, 1689, 478) which are mis-labeled.

**Comment by skdreier** [comment]
TADA List (done on Th 01/16)
- We wrote a loop that successfully added a column for each justification category and appended the outputs for each justification together, yielding 1339 rows of document/justification category (which cross-checked w Nvivo is correct). (#1,4)
- We parsed out each unique reference in a given page to the same justification category code and stored these as a list within a column (#2)
- Fixed the image # for those (PREM 15 1010 1689 478) which were mis-labeled (NOTE: 1010/1689 and 487 have overlapping image numbers (1010/1689: IMGs 001-205; 487: IMGs 001-258). This may be a problem later if we use IMG as a unique identifier. (#5)

TO DO list:
